apis/nucleus/src/object/navigation/navigation.js

```diff
@@ -0,0 +1,70 @@
+import eventmixin from '../../selections/event-mixin';
+
+export default function createNavigationApi(halo, store) {
+  const State = {};
+
+  /**
+   * @class Navigation
+   */
+  const navigationAPI = /** @lends Navigation# */ {
+    /**
+     * Navigate to the supplied sheet
+     * @param {string} sheetId Id of the sheet to navigate to
+     */
+    goToSheet: async (sheetId) => {
+      if (!State.sheetRef) {
+        return;
+      }
+      const { modelStore, rpcRequestModelStore } = store;
+      const key = `${sheetId}`;
+      let rpc = rpcRequestModelStore.get(key);
+      if (!rpc) {
+        rpc = halo.app.getObject(sheetId);
+        rpcRequestModelStore.set(key, rpc);
+      }
+      let model;
+      try {
+        model = await rpc;
+        if (model.genericType !== 'sheet') {
+          return;
+        }
+      } catch (e) {
+        return;
+      }
+      modelStore.set(key, model);
+      State.sheetRef?.current?.setModel?.(model);
+    },
+    /**
+     * @private
+     * Set the current sheet id
+     * @param {string} sheetId Id of the current sheet
+     */
+    setCurrentSheetId: (sheetId) => {
+      if (State.sheetId === sheetId) {
+        return;
+      }
+      State.sheetId = sheetId;
+      navigationAPI.emit('currentSheetIdChange');
```

Comment:
think we should use a simpler name, like "sheetChanged", there is not much reason to have an ID change if the sheet itself didn't change.